src/ray/core_worker/experimental_mutable_object_manager.cc

```diff
@@ -290,12 +291,14 @@ Status MutableObjectManager::ReadAcquire(const ObjectID &object_id,
   size_t total_size = channel->mutable_object->header->data_size +
                       channel->mutable_object->header->metadata_size;
   RAY_CHECK_LE(static_cast<int64_t>(total_size), channel->mutable_object->allocated_size);
+  std::shared_ptr<ChannelBuffer> channel_buffer = std::make_shared<ChannelBuffer>(
+      shared_from_this(), channel->mutable_object->buffer, object_id);
```

Comment:
Not sure that we need this. Will the MutableObjectManager ever be deallocated in the MutableObjectProvider before shutdown?

Reply:
Did you mean why we need a pointer to `MutableObjectManager`?
`ChannelBuffer` needs a pointer to `MutableObjectManager` so that it can call `ReadRelease()` on it.
And we use a shared_ptr of `MutableObjectManager` in `MutableObjectProvider` so all the pointers point to a single `MutableObjectManager`.